### PR TITLE
upstream merge 7/9

### PR DESCRIPTION
**Pull request by deltanedas**
## Notable additions
- uplink discounts are real, deltav and nyano listings updated for it
- atmos alerts computer is now functional
- ai is less buggy
- ghost role rule updates, deltav ones still used where they override things like spiders and slimes

## Notable exclusions
- removed instant death snail from the event
- mysta keeps the binary key until ai is rolled out
- vendors no longer have grappling gun, justice helm or red neck gaiters

**Changelog**
:cl:
- add: Merged upstream changes from last week.

### CODE REVIEW

File: Resources/Prototypes/Catalog/VendingMachines/Inventories/salvage.yml [hunk 1]
```diff
@@ -9,6 +9,7 @@
     Floodlight: 2 # DeltaV - upstream removed them
     HandheldGPSBasic: 2
     RadioHandheld: 2
+    WeaponGrapplingGun: 2
```
Comment (TadJohnson00) [inline Resources/Prototypes/Catalog/VendingMachines/Inventories/salvage.yml]: Following review, we'd appreciate this commented out. Grapple guns break the physics too much and may be pending removal from research also.

File: Resources/Prototypes/Catalog/VendingMachines/Inventories/secdrobe.yml [hunk 2]
```diff
@@ -9,6 +9,7 @@
     ClothingHeadHatBeretSecurity: 3
 ##    ClothingHeadHatSecsoft: 3 # DeltaV - removed from SecDrobe due to inconsistency with art direction
     ClothingHeadBandRed: 3
+    ClothingMaskNeckGaiterRed: 3
```
Comment (TadJohnson00) [inline Resources/Prototypes/Catalog/VendingMachines/Inventories/secdrobe.yml]: Please comment out

File: Resources/Prototypes/Catalog/VendingMachines/Inventories/secdrobe.yml [hunk 3]
```diff
@@ -26,6 +27,7 @@
     ClothingEyesBlindfold: 1
     ClothingShoesBootsCombat: 1
     ClothingShoesBootsWinterSec: 2
+    ClothingHeadHelmetJustice: 1
```
Comment (TadJohnson00) [inline Resources/Prototypes/Catalog/VendingMachines/Inventories/secdrobe.yml]: Please also comment out

**Review by TadJohnson00 — CHANGES_REQUESTED** [review]
Taking a brief peep through the commits, apparently these were the only lines that raised issues.

**Comment by IamVelcroboy** [comment]
Really curious about the action bar change in mapping after you load a few maps....will there be any screen left? 

(Side note: I should just PR the load acts files so everyone can ignore that anyway)

**Review by TadJohnson00 — APPROVED** [review]
Requested changes done, direction stamp of approval given.

**Pull request merged**